spec/features/saml/saml_spec.rb

```diff
@@ -235,6 +233,115 @@ class MockSession; end
     end
   end
 
+  context 'with an SP configured to receive verified attributes' do
+    context 'with a proofed user' do
+      let(:pii) { { phone: '+12025555555', ssn: '111111111', dob: '01/01/1941' } }
+      let(:user) { create(:profile, :active, :verified, pii: pii).user }
+
+      scenario 'sign in flow with user authorizing SP' do
+        visit_idp_from_saml_sp_with_ial2
+        sign_in_live_with_2fa(user)
+        click_submit_default
+        click_agree_and_continue
+        click_submit_default_twice
+
+        xmldoc = SamlResponseDoc.new('feature', 'response_assertion')
+        expect(xmldoc.attribute_value_for(:ial)).to eq(
+          Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF,
+        )
+
+        expect { xmldoc.attribute_value_for(:ssn) }.not_to raise_exception
+        expect(xmldoc.attribute_value_for(:ssn)).to eq('111111111')
+
+        sp_return_logs = SpReturnLog.where(user_id: user.id)
+        expect(sp_return_logs.count).to eq(1)
+        expect(sp_return_logs.first.ial).to eq(2)
+      end
+
+      scenario 'enforces reauthentication when ForceAuthn = true in SAMLRequest' do
+        saml_request_overrides = {
+          issuer: sp1_issuer,
+          authn_context: [
+            Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF,
+            "#{Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF}first_name:last_name email, ssn",
+            "#{Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF}phone",
+          ],
+          force_authn: true,
+          security: {
+            embed_sign: false,
+          }
+        }
+        # start with an active user session
+        sign_in_live_with_2fa(user)
+
+        # visit from SP with force_authn: true
+        visit_saml_authn_request_url(overrides: saml_request_overrides)
+        expect(page.has_content?(
```

Comment:
**nitpick:** I think you can replace this with `expect(page).to have_content(...)`

Reply:
Great point! Fixed in https://github.com/18F/identity-idp/pull/7616/commits/114280bea2a9b129da53a143317111a7f234fbdd